include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,226 @@
+  auto executor_np = dlaf::getNpExecutor<Backend::MC>();
+
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mv = mat_v.nrTiles().rows();
+  const SizeType nv = mat_v.nrTiles().cols();
+  const SizeType mb = mat_v.blockSize().rows();
+  const SizeType nb = mat_v.blockSize().cols();
+  const SizeType ms = mat_v.size().rows();
+  const SizeType ns = mat_v.size().cols();
+
+  // Matrix T
+  int tottaus;
+  if (ms < mb || ms == 0 || nv == 0)
+    tottaus = 0;
+  else
+    tottaus = (ms / mb - 1) * mb + ms % mb;
+
+  if (tottaus == 0)
+    return;
+
+  LocalElementSize sizeT(tottaus, tottaus);
+  TileElementSize blockSizeT(mb, mb);
+  Matrix<T, Device::CPU> mat_t(sizeT, blockSizeT);
+
+  Matrix<T, Device::CPU> mat_vv({mat_v.size().rows(), mb}, mat_v.blockSize());
+  Matrix<T, Device::CPU> mat_w({mat_v.size().rows(), mb}, mat_v.blockSize());
+  Matrix<T, Device::CPU> mat_w2({mb, mat_c.size().cols()}, mat_c.blockSize());
+
+  SizeType last_mb;
+  if (mat_v.blockSize().cols() == 1) {
+    last_mb = 1;
+  }
+  else if (mat_v.size().cols()) {
+    if (mat_v.size().cols() % mat_v.blockSize().cols() == 0)
+      last_mb = mat_v.blockSize().cols();
+    else
+      last_mb = mat_v.size().cols() % mat_v.blockSize().cols();
+  }
+
+  // Specific for V matrix layout where last column of tiles is empty
+  const SizeType last_reflector_idx = mat_v.nrTiles().cols() - 2;
+
+  for (SizeType k = last_reflector_idx; k >= 0; --k) {
+    bool is_last = (k == last_reflector_idx) ? true : false;
+
+    for (SizeType i = k + 1; i < mat_v.nrTiles().rows(); ++i) {
+      // Copy V panel into VV
+      copySingleTile(mat_v.read(LocalTileIndex(i, k)), mat_vv(LocalTileIndex(i, 0)));
+
+      // Setting VV
+      auto tile_i0 = mat_vv(LocalTileIndex(i, 0));
+      if (i <= k) {
+        hpx::dataflow(hpx::launch::sync, unwrapping(tile::set0<T>), std::move(tile_i0));
+      }
+      else if (i == k + 1) {
+        hpx::dataflow(hpx::launch::sync, unwrapping(tile::laset<T>), lapack::MatrixType::Upper, 0.f, 1.f,
+                      std::move(tile_i0));
+      }
+
+      // Copy VV into W
+      copySingleTile(mat_vv.read(LocalTileIndex(i, 0)), mat_w(LocalTileIndex(i, 0)));
+    }
+
+    // Reset W2 to zero
+    setZero(mat_w2);
+
+    // TODO: instead of using a full matrix, choose a "column" matrix. The problem is that last tile
+    // should be square but may have different size.
+    const GlobalTileIndex v_start{k + 1, k};
+    auto taus_panel = taus[k];
+    const SizeType taupan = (is_last) ? last_mb : mat_v.blockSize().cols();
+    dlaf::factorization::internal::computeTFactor<Backend::MC>(taupan, mat_v, v_start, taus_panel,
+                                                               mat_t(LocalTileIndex{k, k}));
+
+    for (SizeType i = k + 1; i < m; ++i) {
+      auto kk = LocalTileIndex{k, k};
+      // WH = V T
+      auto ik = LocalTileIndex{i, 0};

```

Comment:
why is it called `ik` and it is initialized `{i, 0}`?

Reply:
This is because in the algorithm it should be `ik` element, but in our "auxiliary" matrix we are just copying one column at a time. I've maintained the `ik` name in order not to loose the original meaning. 
However, if you think it is better to call it `i0` I will modify this and the following instances! What do you think?

Reply:
I suggest to use the panel, which deals correctly with indices.